contrib/src/main/java/com/datatorrent/contrib/avro/PojoToAvro.java

```
@@ -0,0 +1,264 @@
+  }
+
+  /**
+   * This method generates the getters for provided field of a given class
+   * 
+   * @return Getter
+   */
+  private Getter<?, ?> generateGettersForField(Class<?> cls, String inputFieldName)
+      throws NoSuchFieldException, SecurityException
+  {
+    java.lang.reflect.Field f = cls.getDeclaredField(inputFieldName);
+    Class<?> c = ClassUtils.primitiveToWrapper(f.getType());
+
+    Getter<?, ?> classGetter = PojoUtils.createGetter(cls, inputFieldName, c);
+
+    return classGetter;
+  }
+
+  /**
+   * Initializes the list of columns in POJO based on the names from schema
+   */
+  private void initializeColumnMap(Schema schema)
+  {
+    setColumnNames(schema.getFields());
+
+    keyMethodMap = new ArrayList<Getter>();
+    for (int i = 0; i < getColumnNames().size(); i++) {
+      try {
+        keyMethodMap.add(generateGettersForField(cls, getColumnNames().get(i).name()));
+      } catch (NoSuchFieldException | SecurityException e) {
+        throw new RuntimeException("Failed to initialize pojo class getters for field: ", e);
+      }
+    }
+  }
+
+  @InputPortFieldAnnotation(optional = false, schemaRequired = true)
+  public final transient DefaultInputPort<Object> data = new DefaultInputPort<Object>()
+  {
+
+    @Override
+    public void setup(PortContext context)
+    {
+      cls = context.getValue(Context.PortContext.TUPLE_CLASS);
+
+      try {
+        parseSchema();
+        initializeColumnMap(getSchema());
+      } catch (IOException e) {
+        LOG.error("Exception in parsing schema -" + e.getMessage());
+      }
+    }
+
+    @Override
+    public void process(Object tuple)
+    {
+      processTuple(tuple);
+    }
+
+  };
+
+  /**
+   * Converts incoming tuples into Generic records
+   */
+  protected void processTuple(Object tuple)
+  {
+    GenericRecord record = null;
+
+    try {
+      record = getGenericRecord(tuple);
+    } catch (Exception e) {
+      LOG.error("Exception in parsing record");
```

Please don't hide the exception callstack.. Please log it.